src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,27 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleConfirm }) {
+    super(popupSelector);
+    this._confirmButton = this._popupElement.querySelector(
+      ".modal__save-button"
+    );
+    this._handleConfirm = handleConfirm;
+  }
+
+  open(card, cardId) {
```

Comment:
Needs correction: `PopupWithConfirmation` is a general class. It can be used for any confirmation, not only card delete confirmation. This class should have a setter for `_handleConfirm`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
   this._confirmButton = ...
}

setAction(action) { 
  this._handleConfirm = action; 
} 

this._confirmButton.addEventListener("click", (evt) => { 
  this._handleConfirm(); 
})
```

Reply:
Needs correction: this class should not know anything about cards